AdminUi/src/AdminUi/ClientApp/src/app/components/quotas/assign-quotas-dialog/assign-quotas-dialog.component.ts

```diff
@@ -62,6 +64,56 @@ export class AssignQuotasDialogComponent {
         });
     }
 
+    public createQuota(quotaData: AssignQuotaData): void {
+        this.loading = true;
+
+        if (this.data.address) {
+            this.createIdentityQuota(this.data.address, quotaData);
+        } else if (this.data.tierId) {
+            this.createTierQuota(this.data.tierId, quotaData);
+        }
+    }
+
+    private createIdentityQuota(address: string, quotaData: AssignQuotaData) {
+        const createQuotaRequest = {
+            metricKey: quotaData.metricKey,
+            max: quotaData.max,
+            period: quotaData.period
+        } as CreateQuotaForIdentityRequest;
+
+        this.quotasService.createIdentityQuota(createQuotaRequest, address).subscribe({
+            next: (_: HttpResponseEnvelope<IdentityQuota>) => {
+                this.snackBar.open("Successfully assigned quota.", "Dismiss", {
+                    duration: 4000,
+                    verticalPosition: "top",
+                    horizontalPosition: "center"
+                });
+                this.dialogRef.close(true);
+            },
+            complete: () => {
+                this.loading = false;
+            },
+            error: (err: any) => {
+                this.loading = false;
+                this.errorMessage = err.error?.error?.message ?? err.message;
+            }
+        });
+    }
+
+    private createTierQuota(tierId: string, quotaData: AssignQuotaData) {
+        this.quotasService.createTierQuota(quotaData, tierId).subscribe({
+            next: () => {
+                this.snackBar.open("Successfully assigned quota.", "Dismiss");
+                this.dialogRef.close(true);
+            },
+            complete: () => (this.loading = false),
+            error: (err: any) => {
+                this.loading = false;
+                this.errorMessage = err.error?.error?.message ?? err.message;
+            }
+        });
+    }
```

Comment:
This dialog should be reusable. Currently it is not, because you have to change it every time you want to use it somewhere. What was wrong with the solution as it was before?

Reply:
I see.. I made it not reusable.. I'll revert my changes and try a different solution. :)